pkg/controller/vsphere/reconciler_test.go

```diff
@@ -161,6 +163,23 @@ func TestClone(t *testing.T) {
 			cloneVM:     true,
 			machineName: "test1",
 		},
+		{
+			testCase: "fail on disc resize down",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				CredentialsSecret: &corev1.LocalObjectReference{
+					Name: "test",
+				},
+				Workspace: &vsphereapi.Workspace{
+					Server: server.URL.Host,
+				},
+				DiskGiB:  0,
```

Comment:
Can we make this non-zero or add a separate case? I would expect a zero value to be caught earlier or maybe even produce a different error. While it may not now, we may change this later and that would affect that validity of this test.

Reply:
Done.